File: java/src/test/java/io/binac/leetcode/ValidPalindromeTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class ValidPalindromeTest {
    private static final ValidPalindrome.Solution1 SOLUTION1 = new ValidPalindrome.Solution1();

    private final String S1 = "A man, a plan, a canal: Panama";
    private final boolean EXPECTED1 = true;

    private final String S2 = "race a car";
    private final boolean EXPECTED2 = false;

    private final String S3 = " ";
    private final boolean EXPECTED3 = true;

    @Test
    void test1() {
        assertEquals(EXPECTED1, SOLUTION1.isPalindrome(S1));
        assertEquals(EXPECTED2, SOLUTION1.isPalindrome(S2));
        assertEquals(EXPECTED3, SOLUTION1.isPalindrome(S3));
    }
}

File: cpp/unittests/ValidPalindromeTest.cpp
#include "ValidPalindrome.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(ValidPalindrome, Solution125_1) {
  EXPECT_TRUE(Solution125_1().isPalindrome("A man, a plan, a canal: Panama"));
  EXPECT_FALSE(Solution125_1().isPalindrome("race a car"));
}